src/libs/geotagging.c

```diff
@@ -1276,82 +1273,75 @@ static void _display_datetime(dt_lib_datetime_t *dtw, const time_t datetime,
 }
 
 // read the current date/time and make correction (field under/overflow)
-static time_t _read_datetime_entry(dt_lib_module_t *self)
+static GDateTime *_read_datetime_entry(dt_lib_module_t *self)
 {
   dt_lib_geotagging_t *d = (dt_lib_geotagging_t *)self->data;
-  struct tm dt = {0}; struct tm dt2 = {0};
-  dt.tm_year = dt2.tm_year = atol(gtk_entry_get_text(GTK_ENTRY(d->dt.widget[0])));
-  dt.tm_mon = dt2.tm_mon = atol(gtk_entry_get_text(GTK_ENTRY(d->dt.widget[1])));
-  dt.tm_mday = dt2.tm_mday = atol(gtk_entry_get_text(GTK_ENTRY(d->dt.widget[2])));
-  dt.tm_hour = dt2.tm_hour = atol(gtk_entry_get_text(GTK_ENTRY(d->dt.widget[3])));
-  dt.tm_min = dt2.tm_min = atol(gtk_entry_get_text(GTK_ENTRY(d->dt.widget[4])));
-  dt.tm_sec = dt2.tm_sec = atol(gtk_entry_get_text(GTK_ENTRY(d->dt.widget[5])));
-  dt.tm_year -= 1900; dt2.tm_year -= 1900;
-  dt.tm_mon --; dt2.tm_mon --;
-  dt.tm_isdst = -1;
-
-  time_t datetime = mktime(&dt);
-  // mktime recalculate the different elements - for example 31/02
-  // if difference display the new date
-  if(datetime != -1 &&
-     (dt.tm_year != dt2.tm_year || dt.tm_mon != dt2.tm_mon ||  dt.tm_mday != dt2.tm_mday ||
-      dt.tm_hour != dt2.tm_hour || dt.tm_min != dt2.tm_min || dt.tm_sec != dt2.tm_sec))
-  {
-    _display_datetime(&d->dt, datetime, TRUE, self);
-  }
-  return datetime;
+
+  const int year = atol(gtk_entry_get_text(GTK_ENTRY(d->dt.widget[0])));
```

Comment:
any reason to use `atol()` to fill `int` vars?

Reply:
I haven't any good reason, really. Thanks for pinpointing this.